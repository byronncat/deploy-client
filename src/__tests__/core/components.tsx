import { Sidebar } from '@core';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

describe('component', () => {
  describe('sidebar', () => {
    it('should render correctly', () => {
      const { container } = render(
        <BrowserRouter>
          <Sidebar />
        </BrowserRouter>,
      );
      expect(container).toMatchSnapshot();
    });
  });
});
