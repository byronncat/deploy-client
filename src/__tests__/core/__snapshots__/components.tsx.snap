// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component sidebar should render correctly 1`] = `
<div>
  <nav
    class="h-full bg-white/[.08] d-flex flex-column transition-all duration-300 w-64"
  >
    <a
      class="brand my-10 flex justify-center items-center"
      href="/"
    >
      <img
        alt="logo"
        class="w-12 h-12"
        src="logo.svg"
      />
      <span
        class="brand-name ms-2 font-semibold text-2xl tracking-wider transition-all duration-700 block"
      >
        Bygram
      </span>
    </a>
    <ul
      class="flex flex-col items-center"
    >
      <li
        aria-current="page"
        class="group mb-3 h-12  transition-all duration-300 w-48"
      >
        <a
          class="w-full h-full py-2 rounded flex items-center capitalize w-48 ps-6 text-white/[.6] hover:text-white hover:bg-cerise-700/[.4]"
          href="/"
          id="home"
        >
          <svg
            class="size-5 transition-all duration-300 fill-white/[.6] group-hover:fill-white"
            fill="white"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M575.8 255.5c0 18-15 32.1-32 32.1h-32l.7 160.2c0 2.7-.2 5.4-.5 8.1V472c0 22.1-17.9 40-40 40H456c-1.1 0-2.2 0-3.3-.1c-1.4 .1-2.8 .1-4.2 .1H416 392c-22.1 0-40-17.9-40-40V448 384c0-17.7-14.3-32-32-32H256c-17.7 0-32 14.3-32 32v64 24c0 22.1-17.9 40-40 40H160 128.1c-1.5 0-3-.1-4.5-.2c-1.2 .1-2.4 .2-3.6 .2H104c-22.1 0-40-17.9-40-40V360c0-.9 0-1.9 .1-2.8V287.6H32c-18 0-32-14-32-32.1c0-9 3-17 10-24L266.4 8c7-7 15-8 22-8s15 2 21 7L564.8 231.5c8 7 12 15 11 24z"
            />
          </svg>
          <p
            class="ms-3 whitespace-nowrap transition-all duration-300 block"
          >
            home
          </p>
        </a>
      </li>
      <li
        aria-current="page"
        class="group mb-3 h-12  transition-all duration-300 w-48"
      >
        <a
          class="w-full h-full py-2 rounded flex items-center capitalize w-48 ps-6 text-white/[.6] hover:text-white hover:bg-cerise-700/[.4]"
          href="/"
          id="search"
        >
          <svg
            class="size-5 transition-all duration-300 fill-white/[.6] group-hover:fill-white"
            fill="white"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
            />
          </svg>
          <p
            class="ms-3 whitespace-nowrap transition-all duration-300 block"
          >
            search
          </p>
        </a>
      </li>
      <li
        aria-current="page"
        class="group mb-3 h-12  transition-all duration-300 w-48"
      >
        <a
          class="w-full h-full py-2 rounded flex items-center capitalize w-48 ps-6 text-white/[.6] hover:text-white hover:bg-cerise-700/[.4]"
          href="/explore"
          id="explore"
        >
          <svg
            class="size-5 transition-all duration-300 fill-white/[.6] group-hover:fill-white"
            fill="white"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464 256A208 208 0 1 0 48 256a208 208 0 1 0 416 0zM0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256zm306.7 69.1L162.4 380.6c-19.4 7.5-38.5-11.6-31-31l55.5-144.3c3.3-8.5 9.9-15.1 18.4-18.4l144.3-55.5c19.4-7.5 38.5 11.6 31 31L325.1 306.7c-3.2 8.5-9.9 15.1-18.4 18.4zM288 256a32 32 0 1 0 -64 0 32 32 0 1 0 64 0z"
            />
          </svg>
          <p
            class="ms-3 whitespace-nowrap transition-all duration-300 block"
          >
            explore
          </p>
        </a>
      </li>
      <li
        aria-current="page"
        class="group mb-3 h-12  transition-all duration-300 w-48"
      >
        <a
          class="w-full h-full py-2 rounded flex items-center capitalize w-48 ps-6 text-white/[.6] hover:text-white hover:bg-cerise-700/[.4]"
          href="/"
          id="create"
        >
          <svg
            class="size-5 transition-all duration-300 fill-white/[.6] group-hover:fill-white"
            fill="white"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M64 80c-8.8 0-16 7.2-16 16V416c0 8.8 7.2 16 16 16H384c8.8 0 16-7.2 16-16V96c0-8.8-7.2-16-16-16H64zM0 96C0 60.7 28.7 32 64 32H384c35.3 0 64 28.7 64 64V416c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V96zM200 344V280H136c-13.3 0-24-10.7-24-24s10.7-24 24-24h64V168c0-13.3 10.7-24 24-24s24 10.7 24 24v64h64c13.3 0 24 10.7 24 24s-10.7 24-24 24H248v64c0 13.3-10.7 24-24 24s-24-10.7-24-24z"
            />
          </svg>
          <p
            class="ms-3 whitespace-nowrap transition-all duration-300 block"
          >
            create
          </p>
        </a>
      </li>
      <li
        aria-current="page"
        class="group mb-3 h-12  transition-all duration-300 w-48"
      >
        <a
          class="w-full h-full py-2 rounded flex items-center capitalize w-48 ps-6 text-white/[.6] hover:text-white hover:bg-cerise-700/[.4]"
          href="/profile/undefined"
          id="profile"
        >
          <svg
            class="size-5 transition-all duration-300 fill-white/[.6] group-hover:fill-white"
            fill="white"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 256A128 128 0 1 0 224 0a128 128 0 1 0 0 256zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304H178.3z"
            />
          </svg>
          <p
            class="ms-3 whitespace-nowrap transition-all duration-300 block"
          >
            profile
          </p>
        </a>
      </li>
      <li
        aria-current="page"
        class="group mb-3 h-12  transition-all duration-300 w-48"
      >
        <a
          class="w-full h-full py-2 rounded flex items-center capitalize w-48 ps-6 text-white/[.6] hover:text-white hover:bg-cerise-700/[.4]"
          href="/login"
          id="logout"
        >
          <svg
            class="size-5 transition-all duration-300 fill-white/[.6] group-hover:fill-white"
            fill="white"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M377.9 105.9L500.7 228.7c7.2 7.2 11.3 17.1 11.3 27.3s-4.1 20.1-11.3 27.3L377.9 406.1c-6.4 6.4-15 9.9-24 9.9c-18.7 0-33.9-15.2-33.9-33.9l0-62.1-128 0c-17.7 0-32-14.3-32-32l0-64c0-17.7 14.3-32 32-32l128 0 0-62.1c0-18.7 15.2-33.9 33.9-33.9c9 0 17.6 3.6 24 9.9zM160 96L96 96c-17.7 0-32 14.3-32 32l0 256c0 17.7 14.3 32 32 32l64 0c17.7 0 32 14.3 32 32s-14.3 32-32 32l-64 0c-53 0-96-43-96-96L0 128C0 75 43 32 96 32l64 0c17.7 0 32 14.3 32 32s-14.3 32-32 32z"
            />
          </svg>
          <p
            class="ms-3 whitespace-nowrap transition-all duration-300 block"
          >
            logout
          </p>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
